source/D2Common/src/D2Skills.cpp

```diff
@@ -1825,41 +1825,34 @@ BOOL __fastcall sub_6FDB1130(D2UnitStrc* pItem, D2UnitStrc* pUnit, D2SkillsTxt*
 //TODO: Check name
 BOOL __fastcall D2Common_SKILLS_CheckShapeRestriction_6FDB1380(D2UnitStrc* pUnit, D2SkillStrc* pSkill)
 {
-	int nCounter = 0;
+	if (pUnit == nullptr || pSkill == nullptr || pSkill->pSkillsTxt == nullptr) {
+		return FALSE;
+	}
 
-	if (pUnit && pSkill)
+	if (pSkill->pSkillsTxt->nRestrict == 0)
```

Comment:
Might be a good idea to introduce a local variable pSkillsTxt to avoid all the pSkill->pSkillsTxt